azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -27,12 +39,31 @@ class PolicyError(AgentError):
     """
 
 
-class PolicyEngine(object):
+class PolicyInvalidError(AgentError):
+    """
+    Error raised if user-provided policy is invalid.
+    """
+    def __init__(self, msg, inner=None):
+        msg = "Customer-provided policy file ('{0}') is invalid, please correct the following error: {1}".format(_CUSTOM_POLICY_PATH, msg)
+        super(PolicyInvalidError, self).__init__(msg, inner)
+
+
+class _PolicyEngine(object):
     """
     Implements base policy engine API.
     """
-    @classmethod
-    def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+    def __init__(self):
+        # Policy enablement is checked and policy file is read once per initialization (per goal state).
+        # The same policy should be enforced even if the policy file is deleted/changed during a single goal state processing.
+        self._policy_enforcement_enabled = self.__get_policy_enforcement_enabled()
+        if not self.policy_enforcement_enabled:
+            self._log_policy_event(msg="Policy enforcement is not enabled.")
```

Comment:
On most VMs, the policy file will not exist and this message won't make much sense. Maybe log when enforcing, rather than when not enforcing? (and include the path to the policy file in the message)

Reply:
Removed this and added a log message only when enforcing